### ISSUE TITLE
Fix fakewebsitebuster and scamadviser broken pipe errors

### ISSUE THREAD

**Issue by jarelllama** [jarelllama/Scam-Blocklist#349]
Error:
``` console
grep: write error: Broken pipe
```

Scamadviser domain retrieval is also broken:
```
10:08:32 27-04-24,invalid,10-most-faked-products-sold-online,scamadviser.com
```

These two sources use similar regex:

``` bash
source_fakewebsitebuster() {
    ...
    curl -sS --retry 2 --retry-all-errors "${url}/" \
        | grep -oE 'rel="bookmark">.*</a></h2>' \
        | grep -oE "([0-9]|[A-Z])${DOMAIN_REGEX}" \
        | head -n 50 > "$results_file"  # Keep only newest 50 results
}

...

source_scamadviser() {
    ....
    # URL globbing added after https://github.com/T145/black-mirror/issues/179
    # Trailing slash intentionally omitted
    curl -sSZ --retry 2 --retry-all-errors "${url}?p=[1-15]" \
        | grep -oE '<h2 class="mb-0">.*</h2>' \
        | grep -oE "([0-9]|[A-Z])${DOMAIN_REGEX}" > "$results_file"
}
```

**Comment by jarelllama** [comment]
Due to updated regex for matching both example.com and example-com. Will revert.

https://github.com/jarelllama/Scam-Blocklist/blob/91eef19407249deb045d931d821660755631d2c1/scripts/retrieve_domains.sh#L23-L24

Still have to find the root cause for the pipe errors.

**Comment by jarelllama** [comment]
The grep pipe error appears to be because of the use of wildcard matching in my regex:
``` bash
grep -oE '<h2 class="mb-0">.*</h2>' \
| grep -oE "([0-9]|[A-Z])${DOMAIN_REGEX}" \
```
This is done because the ScamAdviser page has all the results in one long line
``` html
<h2 class="mb-0">Is Zuoped.com Legit or a Scam?</h2>...
```
My guess is because of the long string being written to the pipe, the read end of the second grep closes before the write end of the first grep finishes.
https://superuser.com/questions/554855/how-can-i-fix-a-broken-pipe-error/554896#554896
